subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesMultiTargetIntegrationTest.groovy

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testsuites
+
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestFramework
+import org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+
+import static org.junit.Assume.assumeNotNull
+
+class TestSuitesMultiTargetIntegrationTest extends AbstractIntegrationSpec {
+    String jdkVersion
+
+    def setup() {
+        jdkVersion = AvailableJavaHomes.getAvailableJdk {
+            // Include anything that isn't Java 8, and will still run Java 8 classes
+            it.languageVersion.isJava9Compatible()
+        }?.javaVersion?.majorVersion
+        assumeNotNull(jdkVersion)
+    }
+
+    def "multiple targets can be used"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
```

Comment:
Instead of relying on the other JDK being compatible with JDK 8, we should make the build target JDK 8.  That way any version of the JDK we pick for the tests will just work.

e.g., something like:
```
           java {
                sourceCompatibility = 8
                targetCompatibility = 8
            }
```